apps/ledger-live-mobile/src/components/GradientContainer.tsx

```diff
@@ -20,19 +21,43 @@ export default function GradientContainer({
   children,
   startOpacity = 1,
   endOpacity = 0,
+  direction = "top-to-bottom",
 }: Props) {
   const { colors } = useTheme();
 
+  let x1 = "0%";
+  let y1 = "0%";
+  let x2 = "0%";
+  let y2 = "100%";
+
+  switch (direction) {
+    case "left-to-right":
+      x2 = "100%";
+      y2 = "0%";
+      break;
+    case "bottom-to-top":
+      y1 = "100%";
+      y2 = "0%";
+      break;
+    case "right-to-left":
+      x1 = "100%";
+      x2 = "0%";
+      break;
+  }
+
```

Comment:
Can you extract this logic outside of the component, please?
Maybe we can have something like `const positions = getPositions(direction)`
Then we can even spread the result in props:
```
<LinearGradient
            {...positions}
            id="myGradient"
            gradientUnits="userSpaceOnUse"
          >
```
That way we don't redefine the logic at each rerender + it improves readability of the component.

Reply:
Done ✅ 
LTMK if it's what u meant